apps/arweave/src/ar_http_iface_server.erl

```diff
@@ -65,157 +64,34 @@ start() ->
-			ok;
+	case CertfilePath of
 		not_set ->
-			ok
-	end.
-
-start_http_gateway_listener(Config) ->
-	Domain = Config#config.gateway_domain,
-	CustomDomains = Config#config.gateway_custom_domains,
-	TransportOpts = #{
-		max_connections => 100,
-		num_acceptors => 100,
-		socket_opts => [{port, 80}]
-	},
-	ProtocolOpts = protocol_opts([{gateway, Domain, CustomDomains}]),
-	{ok, _} = cowboy:start_clear(ar_http_gateway_listener, TransportOpts, ProtocolOpts),
-	ok.
-
-start_https_gateway_listener(Config) ->
-	PrivDir = code:priv_dir(arweave),
-	Domain = Config#config.gateway_domain,
-	CustomDomains = Config#config.gateway_custom_domains,
-	Dispatch = cowboy_router:compile([{'_', ?HTTP_IFACE_ROUTES}]),
-	HttpIfaceEnv = #{ dispatch => Dispatch },
-	SniHosts = derive_sni_hosts(CustomDomains),
-	TransportOpts = #{
-		max_connections => Config#config.max_gateway_connections,
-		socket_opts => [
-			{port, 443},
-			{certfile, filename:join([PrivDir, "tls/cert.pem"])},
-			{keyfile, filename:join([PrivDir, "tls/key.pem"])},
-			{sni_hosts, SniHosts}
-		] ++ cacertfile_when_present()
-	},
-	ProtocolOpts = protocol_opts([{http_iface, HttpIfaceEnv}, {gateway, Domain, CustomDomains}]),
-	{ok, _} = cowboy:start_tls(ar_https_gateway_listener, TransportOpts, ProtocolOpts),
-	ok.
-
-protocol_opts(List) ->
-	Opts1 = #{ middlewares := Middlewares1, env := Env1 } =
-		case proplists:lookup(http_iface, List) of
-			{http_iface, HttpIfaceEnv} ->
-				#{
-					inactivity_timeout => 120000,
-					idle_timeout => 30000,
-					middlewares => ?HTTP_IFACE_MIDDLEWARES,
-					env => HttpIfaceEnv,
-					metrics_callback => fun prometheus_cowboy2_instrumenter:observe/1,
-					stream_handlers => [cowboy_metrics_h, cowboy_stream_h]
-				};
-			none ->
-				#{ middlewares => [], env => #{} }
-		end,
-	Opts2 =
-		case proplists:lookup(gateway, List) of
-			{gateway, Domain, CustomDomains} ->
-				Opts1#{
-					middlewares := [ar_gateway_middleware | Middlewares1],
-					env := Env1#{
-						gateway => {Domain, CustomDomains},
-						arql_semaphore => gateway_arql
-					}
-				};
-			none ->
-				Opts1#{
-					env := Env1#{ arql_semaphore => arql },
-					metrics_callback => fun collect_http_response_metrics/1
-				}
-		end,
-	Opts2.
-
-derive_sni_hosts(Domains) ->
-	PrivDir = code:priv_dir(arweave),
-	lists:map(fun(Domain) ->
-		SDomain = binary_to_list(Domain),
-		{SDomain, [
-			{certfile, filename:join([PrivDir, "tls", SDomain, "cert.pem"])},
-			{keyfile, filename:join([PrivDir, "tls", SDomain, "key.pem"])}
-		] ++ cacertfile_when_present(SDomain)}
-	end, Domains).
-
-cacertfile_when_present() ->
-	cacertfile_when_present(apex_domain).
-
-cacertfile_when_present(Domain) ->
-	PrivDir = code:priv_dir(arweave),
-	Filename = case Domain of
-		apex_domain ->
-			filename:join([PrivDir, "tls/cacert.pem"]);
-		CustomDomain ->
-			filename:join([PrivDir, "tls", CustomDomain, "cacert.pem"])
+			{ok, _} = cowboy:start_clear(ar_http_iface_listener, TransportOpts, ProtocolOpts);
```

Comment:
This is branch with TLS, right? It looks like the TransportOpts and ProtocolOpts have changed vs. what they were before. Just calling that out - I don't know enough to comment onw hether the changes could cause a problem with the non-TLS flow

Reply:
ok it looks like it's only `max_connections` vs. `max_concurrent_streams` that's changed. Same comment though: I have no idea if that change matters - just flagging it

Reply:
this branch on line 87 is normal http, and that doesn't need any adjustments

Reply:
the latter comment I didn't see, now I see where you're coming from, yes it seems the api changed, or rather, changed even in earlier version, maybe there's backwards compatibility built in.

ProtocolOpts: https://ninenines.eu/docs/en/ranch/2.1/manual/ranch_tcp/

I wonder if it's even removed, checking...

Reply:
ok I'm looking at http2 manual, but we are using http1.1, let me fix that

Reply:
https://ninenines.eu/docs/en/cowboy/2.10/manual/cowboy_http2/

Reply:
reverted, max_connections is a part of the TransportOption as it was. We can research further ranch upgrades and the options needed for seamless migration, for another day.